lib/events/filesessions/fileasync.go

```diff
@@ -486,7 +455,16 @@ func encryptedUploadAggregateIter(in io.Reader, targetSize int, maxSize int) ite
 		return nil
 	}
 
-	return func(yield func([]byte, error) bool) {
+	return func(yieldFn func([]byte, error) bool) {
+		// wrap any yielded errors with sessionError since any failure to
+		// read the file itself should be treated as a corrupted recording
+		yield := func(b []byte, err error) bool {
+			if err != nil {
+				err = sessionError{err}
+			}
+			return yieldFn(b, err)
```

Comment:
Suggestion: return instead of shadowing?
```suggestion
			if err != nil {
				return yieldFn(b, sessionError{err})
			}
			return yieldFn(b, nil)
```